include/saber/inexact.hpp

```diff
@@ -0,0 +1,71 @@
+#ifndef SABER_INEXACT_HPP
+#define SABER_INEXACT_HPP
+#pragma once
+
+// std
+#include <algorithm>
+#include <cmath>
+#include <limits>
+#include <cassert>
+#include <type_traits>
+
+namespace saber {
+
+class Inexact
+{
+public:
+    template<typename T>
+    static bool IsEQ(T inLHS, T inRHS)
+    {
+        EQ equal{inLHS};
+        const bool result = equal(inRHS);
+        return result;
+    }
+
+    /*
+    template<typename T>
+    static bool IsNE(T inLHS, T inRHS)
+    {
+        EQ unequal{inLHS};
+        const bool result = unequal(inRHS);
+        return result;
+    }
+    */
+
+    template<typename T>
+    struct EQ
+    {
+    public:
+        EQ(const T& inLHS) :
+            mLHS{inLHS}
+        {
+            bool assertion = std::is_floating_point<T>::value;
+            assert(assertion);
```

Comment:
`assert()` isn't the right tool for this job...

`assert()` tests for a value at run-time.
instead: you want to test for a type at compile-time.
The compile-time test is: `static_assert(constexpr value, "error string");`

This lets you flag/emit an error message of your choice during compilation.
Instead of the compiler spewing its own when it hits `operator()(const T&)`...
(imagine `T=std::string`... where you try and do `std::abs(mLHS)` where `mLHS` is a `std::string`. Think of all those crazy errors!)